cmd/aresd/cmd/cmd.go

```diff
@@ -56,6 +57,47 @@ type Options struct {
 // Option is for setting option
 type Option func(*Options)
 
+// singleton of AresD instance
+var aresd *AresD
+var once sync.Once
+
+// AresD is a wrapper of original functions for code reuse
+type AresD struct {
+	sync.Mutex
+	// server configuration and options
+	cfg     common.AresServerConfig
+	options *Options
+	// server is http server instance started inside AresD
+	server *http.Server
+	// WaitCond is used to notify other route that the server is started
+	WaitCond *sync.Cond
+}
+
+// NewAresD create singleton of AresD
+func NewAresD(cfg common.AresServerConfig, options *Options) *AresD {
+	once.Do(func() {
```

Comment:
if there won't be multiple threads accessing this method, we don't need to make this a singleton

if only in test suites we will have multi threads sharing one aresd then the singletone should be placed in test suite only

Reply:
this is just the prevention for future use to create multiple instance

Reply:
I think whoever is the user of NewAresD method should control how many instance they want, not the NewAresD method itself.

In this case, the test suite is the user of NewAresD

Reply:
suppose only one aresd can run, while it will need more logic to control it. I will rollback to make it simpler